### ISSUE TITLE
[RL] Add CI to guard on numerics change

### ISSUE THREAD

**Issue by wwwjn** [pytorch/torchtitan#2691]
We should aim for adding the numerics using the same pattern with pre-training side CI:

1. Locally check our model numerics matches vllm's native model
2. Enable deterministic mode. Run with specific task and getting training metrics , eg reward , loss. Record these metrics in a file.
3. Run CI to compare against the loss in the file.

**Comment by yichuan-w** [comment]
I enabled items 2 and 3 in #3474, while item 1 has not been resolved or verified yet.
